### ISSUE TITLE
Add the logging of exceptions

### ISSUE THREAD

**Issue by shpaass** [shpaass/yafc-ce#92]
Use case:
1. A user opens YAFC, but it crashes immediately without any message. It only crashes for the user and not the developers.
2. Since there are no logs that a user can share, the debugging is hard.

To fix that, I propose to write into an exception-log file. Let the exceptions be written down in the file even if they are shown in YAFC. That will allow to see the reason why YAFC crashed client-side if the error was not shown by YAFC itself.

**Comment by shihan42** [comment]
Very good idea and indeed helpful. We could start with just exceptions and than gradually extend the solution to do proper logging (see #54).

**Comment by veger** [comment]
I just had a quick look and there are a lot of places where exceptions are caught and a `Console.Writeline()` is used to log the exception. So I feel the exceptions will be stored to an (exception) log file already when we tackle #54?

Note that for Linux I can see startup exceptions just fine (being logged to the terminal). For example if SDL2 image library is missing I get:
```
Unhandled exception. System.DllNotFoundException: Unable to load shared library 'SDL2_image-2.0.so.0' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libSDL2_image-2.0.so.0: cannot open shared object file: No such file or directory
   at System.Runtime.InteropServices.NativeLibrary.LoadByName(String libraryName, QCallAssembly callingAssembly, Boolean hasDllImportSearchPathFlag, UInt32 dllImportSearchPathFlag, Boolean throwOnError)
   at System.Runtime.InteropServices.NativeLibrary.LoadLibraryByName(String libraryName, Assembly assembly, Nullable`1 searchPath, Boolean throwOnError)
   at System.Runtime.InteropServices.NativeLibrary.Load(String libraryName, Assembly assembly, Nullable`1 searchPath)
   at YAFC.YafcLib.DllResolver(String libraryname, Assembly assembly, Nullable`1 searchpath) in /home/user/yafc/YAFC/YafcLib.cs:line 59
   at System.Runtime.InteropServices.NativeLibrary.LoadLibraryCallbackStub(String libraryName, Assembly assembly, Boolean hasDllImportSearchPathFlags, UInt32 dllImportSearchPathFlags)
   at SDL2.SDL_image.IMG_Init(IMG_InitFlags flags)
   at YAFC.UI.Ui.Start() in /home/user/yafc/YAFCui/Core/Ui.cs:line 33
```

Which is more reason to see this as part of #54

There is an `ExceptionScreen.ShowException()` function, we could log the exception overthere. But I feel that YAFC crashing on startup won't even reach the UI being activated? So logging these exceptions would not help debugging startup issues.

**Comment by shihan42** [comment]
I agree with you. #54 could solve this problem along the way.
I'd volunteer, if you don't object.

**Comment by veger** [comment]
Feel free to pick it up if you want.
I do not have access to a Windows machine, so it would be hard to me to test.

**Comment by DaleStan** [comment]
I think we can close this as fixed by #54
